### ISSUE TITLE
large attachments hang when opening outside of TextSecure

### ISSUE THREAD

**Issue by mlsxlist** [WhisperSystems/Signal-Android#4116]
Someone sent me a large video of about 32 MB size that was downloaded to TextSecure. When I click the attachment icon in the conversation, TextSecure asks me if I want to view secure media. I confirm with YES and am asked which app should be used for the action. I select Google Video Player and  from now on it hangs. I have to kill Galerie to stop the hanging app.

When I save the attachment to the sdcard I can open it without any problem. Only directly opening from TextSecure does not work.

TS 2.27.1 on Android 4.4.4 cyanogenmod on Galaxy S2

**Comment by mlsxlist** [comment]
I did some more research and tests with different attachment sizes. I found that it only appears as if it hangs. Actually for the 32MB video it takes around 4 minutes until Google Video Player started playing it. A video of 2.9 MB needs 22 sec until it starts playing. During that time there is only a black screen with a "Google Video Player" title. If I save the same video 2.9 MB from Textsecure to SD card, the decrypting and saving process only takes a second. So I am wondering why it takes so long for decrypting and playing the video with Google Video Player.
